src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -61,11 +61,23 @@ internal IAggregator[] MapToMetrics(string[] seqKey, object[] seqVal)
             {
                 aggregators.Add(new SumMetricAggregatorLong(this.instrument.Name, this.instrument.Description, this.instrument.Unit, this.instrument.Meter, dt, tags));
             }
+            else if (this.instrument.GetType() == typeof(ObservableCounter<long>)
```

Comment:
This could be slow comparing to a flag in a struct.

Reply:
yes! all this would be moving away from hot path.